src/lib/Store.ts

```diff
@@ -0,0 +1,108 @@
+export interface StoreConstants {
+    DB_VERSION: number;
+    DB_NAME: string;
+    DB_STORE_NAME: string;
+}
+
+export abstract class Store<Info, Entry> {
+    private static INDEXEDDB_IMPLEMENTATION = window.indexedDB;
+
+    private _dbPromise: Promise<IDBDatabase> | null;
+    private _indexedDB: IDBFactory;
+
+    protected constructor() {
+        this._dbPromise = null;
+        this._indexedDB = Store.INDEXEDDB_IMPLEMENTATION;
+    }
+
+    public async get(id: string): Promise<Info | null> {
+        const db = await this.connect();
+        const transaction = db.transaction(this.getConstants().DB_STORE_NAME);
+        const request = transaction.objectStore(this.getConstants().DB_STORE_NAME).get(id);
+        const result = await this._requestAsPromise(request, transaction);
+        return result ? this.fromEntry(result) : result;
+    }
+
+    public async put(value: Info) {
+        const db = await this.connect();
+        const transaction = db.transaction(this.getConstants().DB_STORE_NAME, 'readwrite');
+        const request = transaction.objectStore(this.getConstants().DB_STORE_NAME).put(this.toEntry(value));
+        return this._requestAsPromise(request, transaction);
+    }
+
+    public async remove(id: string) {
+        const db = await this.connect();
+        const transaction = db.transaction(this.getConstants().DB_STORE_NAME, 'readwrite');
+        const request = transaction.objectStore(this.getConstants().DB_STORE_NAME).delete(id);
+        return this._requestAsPromise(request, transaction);
+    }
+
+    public async list(): Promise<Entry[]> {
```

Comment:
Wouldn't it be nicer if this would return `Info[]`?

Reply:
As it is used right now it always expects a `Entry[]` and neither of those entries are ever converted to `Info`.

So I would say it should stay `Entry[]`.